### PR TITLE
in_toto_attestation/v1: fix type hints

### LINKED ISSUE #300: `in_toto_attestation` broken on Python < 3.9
It looks like `in_toto_attestation` is marked for Python 3.7+ but is using annotation features that were only stabilized in 3.9+, e.g. `type[T]` instead of typing.Type[T]`.

From https://github.com/sigstore/sigstore-python/actions/runs/7134485043/job/19429407875?pr=804:

```console
Traceback (most recent call last):
  File "/home/runner/work/sigstore-python/sigstore-python/env/bin/sigstore", line 5, in <module>
    from sigstore._cli import main
  File "/home/runner/work/sigstore-python/sigstore-python/sigstore/_cli.py", line 54, in <module>
    from sigstore.sign import SigningContext
  File "/home/runner/work/sigstore-python/sigstore-python/sigstore/sign.py", line 55, in <module>
    from in_toto_attestation.v1.statement import Statement
  File "/home/runner/work/sigstore-python/sigstore-python/env/lib/python3.8/site-packages/in_toto_attestation/v1/statement.py", line 4, in <module>
    from in_toto_attestation.v1.resource_descriptor import ResourceDescriptor
  File "/home/runner/work/sigstore-python/sigstore-python/env/lib/python3.8/site-packages/in_toto_attestation/v1/resource_descriptor.py", line 8, in <module>
    class ResourceDescriptor:
  File "/home/runner/work/sigstore-python/sigstore-python/env/lib/python3.8/site-packages/in_toto_attestation/v1/resource_descriptor.py", line [22](https://github.com/sigstore/sigstore-python/actions/runs/7134485043/job/19429407875?pr=804#step:5:23), in ResourceDescriptor
    def copy_from_pb(proto: type[rdpb.ResourceDescriptor]) -> 'ResourceDescriptor':
TypeError: 'type' object is not subscriptable
```

### PR DESCRIPTION
**Pull request by woodruffw**
Fixes #300.

My editor made some formatting changes as well, based on the `black` rules -- let me know if you'd prefer to not have these, and I'll limit this PR to just the type fix.

### CODE REVIEW

File: python/in_toto_attestation/v1/resource_descriptor.py
Comment (woodruffw) [inline python/in_toto_attestation/v1/resource_descriptor.py:33]: NB: I thought a bit about this, and I think this hint was wrong to begin with: `type[T]` is used an annotation for class objects themselves, which isn't what this API takes.

**Review by joshuagl — APPROVED** [review]
Awesome, thanks!

Aside: IMHO we should standardise on `black` code style & lint Python changes with `black` in the CI.

**Comment by woodruffw** [comment]
> Aside: IMHO we should standardise on `black` code style & lint Python changes with `black` in the CI.

Sounds good, I can do this 🙂 

(Unless there's a very strong preference, I like `ruff format` -- it's style compatible with `black`, but about 100x faster.)

**Comment by adityasaky** [comment]
would we set it up in CI to ignore the proto generated files?

**Review by TomHennen — APPROVED** [review]

**Comment by woodruffw** [comment]
Does someone have merge permissions here? I'd like to land this in a release to unblock the work on DSSE support in `sigstore-python` 🙂

**Comment by TomHennen** [comment]
> Does someone have merge permissions here? I'd like to land this in a release to unblock the work on DSSE support in `sigstore-python` 🙂

Done.  I'm never sure what the right etiquette here is.

**Pull request merged**
